### PR TITLE
WIP example of RBAC resource propagation to member clusters for different…

### LINKED ISSUE #526: Support federated clusters in separate namespaces
In our use case we want to run 'fleets' of kubernetes clusters which would be owned by different groups which only have permissions to get / list / update / delete within their fleet. By creating federated cluster resources in separate namespaces in the federation cluster we could leverage RBAC to scope user access to different clusters. 

Currently the federation controller is scoped to one namespace, so we would need to run a separate controller in each federation namespace. That's fine with me but it would be great if it could act as a controller for federated clusters in multiple namespaces (or just all namespaces). Also, currently if you try to deploy the controller helm chart to two different namespaces you get a conflict error:

```
customresourcedefinitions.apiextensions.k8s.io "federatedclusterroles.primitives.federation.k8s.io" already exists
```

Wondering if this is a use case which could be in scope. Happy to contribute a PR if so.

### PR DESCRIPTION
**Pull request by xunpan**
This example is try to help on #526. A cluster scope federation cluster is installed. The RBAC to limit different users access different member clusters within their namespace.

@gyliu513

### CODE REVIEW

File: docs/propagation-rbac-for-member-clusters.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
+This guide shows a way to make a user propagate federated resources to a set of member clusters in kubernetes federation.
+
+`FederatedNamespace` controls which cluster a federated namespace can be propogated to. It also determines which clusters a federated resource of the namespace can be propagated to. `RBAC` can be created to make a user only operate resources in the specified namespace. Here is an example to leverage both `FederatedNamespace` and `RBAC` to limit user propagation rights to a set of member clusters.
```
Comment (poothia) [inline docs/propagation-rbac-for-member-clusters.md]: s/propogated/propagated

File: docs/userguide.md [hunk 2]
```diff
@@ -16,17 +16,19 @@
...
-    - [Check Status of Resources](#check-status-of-resources)
-    - [Update FederatedNamespace Placement](#update-federatednamespace)
-      - [Using Cluster Selector](#using-cluster-selector)
-        - [Neither `spec.placement.clusterNames` nor `spec.placement.clusterSelector` is provided](#neither-specplacementclusternames-nor-specplacementclusterselector-is-provided)
-        - [Both `spec.placement.clusterNames` and `spec.placement.clusterSelector` are provided](#both-specplacementclusternames-and-specplacementclusterselector-are-provided)
-        - [`spec.placement.clusterNames` is not provided, `spec.placement.clusterSelector` is provided but empty](#specplacementclusternames-is-not-provided-specplacementclusterselector-is-provided-but-empty)
-        - [`spec.placementclusterNames` is not provided, `spec.placement.clusterSelector` is provided and not empty](#specplacementclusternames-is-not-provided-specplacementclusterselector-is-provided-and-not-empty)
-    - [Example Cleanup](#example-cleanup)
-    - [Troubleshooting](#troubleshooting)
+    - [Basic Resource Propogation](#basic-resource-propogation)
```
Comment (poothia) [inline docs/userguide.md]: s/Propogation/Propagation

s/basic-resource-propogation/basic-resource-propagation

File: docs/userguide.md [hunk 3]
```diff
@@ -226,14 +228,15 @@ kubefed2 disable <FederatedTypeConfig Name> --delete-from-api
 **WARNING: All custom resources for the type will be removed by this command.**
 
 ## Example
+### Basic Resource Propogation
```
Comment (poothia) [inline docs/userguide.md]: s/Propogation/Propagation

File: docs/propagation-rbac-for-member-clusters.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+This example shows that `user1` only progagates federated resources to `cluster1` and `user2` can propagate federated resources to both `cluster2` and `cluster3`.
```
Comment (poothia) [inline docs/propagation-rbac-for-member-clusters.md]: s/progagates/propagates

**Comment by xunpan** [comment]
This case does not make sense after evalation. So, close this ticket.

With multiple namespaced federation support #631, it is easy to make a user/team uses dedicate clusters for their requirement. Arbitrary member clusters can be added in to different federations for specific users.

**Pull request closed without merging**